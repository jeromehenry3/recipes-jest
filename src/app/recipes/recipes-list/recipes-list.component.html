<div class="row">
  recipes list
</div>
