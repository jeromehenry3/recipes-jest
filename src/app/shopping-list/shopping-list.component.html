<div class="row">
  <div class="col-xs-10">
    <app-shopping-edit></app-shopping-edit>
    <hr />
    <p>ici vient la liste</p>
  </div>
</div>
