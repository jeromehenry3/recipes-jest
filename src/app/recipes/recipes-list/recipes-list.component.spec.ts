import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { RecipesListComponent } from './recipes-list.component';
import { By } from '@angular/platform-browser';
import { ShoppingListComponent } from 'src/app/shopping-list/shopping-list.component';
import { RecipeDetailComponent } from '../recipe-detail/recipe-detail.component';
import { RecipeItemComponent } from './recipe-item/recipe-item.component';
import { MockComponent } from 'ng-mocks';

describe('RecipesListComponent', () => {
  let component: RecipesListComponent;
  let fixture: ComponentFixture<RecipesListComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        RecipesListComponent,
        MockComponent(RecipeItemComponent),
      ],
      providers: []
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(RecipesListComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    const recipesList = fixture.debugElement.nativeElement;
    expect(recipesList).toMatchSnapshot();
  });

  it('should render `RecipeItemComponent`', () => {
    const recipeItem = fixture.debugElement.query(By.directive(RecipeItemComponent));
    expect(recipeItem.componentInstance).toBeTruthy();
  });
});
