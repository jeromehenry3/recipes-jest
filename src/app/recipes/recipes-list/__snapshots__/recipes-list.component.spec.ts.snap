// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesListComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="row"
  >
     recipes list

  </div>
</div>
`;
